src/suffixtree.jl

```diff
@@ -0,0 +1,112 @@
+using Base.Test
+import Base.insert!
+
+abstract SuffixTrie
+
+type SuffixTrieNode <: SuffixTrie
+  value
```

Comment:
For types, any value without a type defaults to `Any`.  It would be better to parameterize both `SuffixTrie` and `SuffixTrieNode` (i.e., as `SuffixTrie{T}` and `SuffixTrieNode{T}`) and make this `value::T`.